permissions-lint/build.gradle.kts

```diff
@@ -83,6 +83,6 @@ dependencies {
 }
 
 java {
-    sourceCompatibility = JavaVersion.VERSION_1_8
-    targetCompatibility = JavaVersion.VERSION_1_8
+    sourceCompatibility = JavaVersion.VERSION_17
+    targetCompatibility = JavaVersion.VERSION_17
```

Comment:
If I understand correctly, this will require consuming projects to build with at least JDK 17, since `permissions-lint` is consumed at Gradle build time.

That SGTM, given that AGP has required that level for a while now.

Reply:
Yeah that's how I understand it as well but I wasn't 100% sure if it was right. Changing it to 1.8 though fails to build